addons/web/static/src/js/views/calendar/calendar_model.js

```diff
@@ -499,9 +499,11 @@ return AbstractModel.extend({
                     fields: self.fieldNames,
                     domain: self.data.domain.concat(self._getRangeDomain()).concat(self._getFilterDomain())
             })
-            .then(function (events) {
+            .then(async function (events) {
                 self._parseServerData(events);
                 self.data.data = _.map(events, self._recordToCalendarEvent.bind(self));
+                var data = await self._calendarEventByAttendee(self.data.data);
+                self.data.data = data.length ? data : self.data.data;
```

Comment:
I would instead extract all the data preparation into a "_getCalendarEventData" method
that returns : `_.map(events, self._recordToCalendarEvent.bind(self));`
and that you could override to do
```
var calendarEventData = super()
calendarEventByAttendeeData = await self._calendarEventByAttendee(calendarEventData);
return calendarEventByAttendeeData.lenght ? calendarEventByAttendeeData : calendarEvent
```
It would avoid to create a _calendarEventByAttendee method that returns nothing in this module